### PR TITLE
feat(vertexai): Multimodal sample with video

### PR DESCRIPTION
**Pull request by Deleplace**
Same code as the multimodal sample, but with the URL of a video.

### CODE REVIEW

**Review by eliben — COMMENTED** [review]
Is this absolutely identical to the exisiting `multimodal` sample, except that the URL points to a video file?

**Comment by Deleplace** [comment]
Yes.
IMO it's fine to have code duplicated in the modules of each samples. 

Factoring would imply putting image+video in the same sample folder, which is fine too.

**Comment by eliben** [comment]
> Yes. IMO it's fine to have code duplicated in the modules of each samples.
> 
> Factoring would imply putting image+video in the same sample folder, which is fine too.

I think maybe we can just show both video or image in the multimodal sample and link to it from both places?
Having identical samples different with just the link may be confusing to users who will be looking for the difference and not finding any

**Comment by Deleplace** [comment]
In this page we want to show a distinct snippet in the two section "Image" and "Video"

https://cloud.google.com/vertex-ai/docs/generative-ai/multimodal/send-multimodal-prompts#gemini-send-multimodal-samples-go

File: vertexai/multimodal-video/multimodal.go [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+	"cloud.google.com/go/vertexai/genai"
+)
+
+func main() {
+	projectID := os.Getenv("GOOGLE_CLOUD_PROJECT")
+	location := "us-central1"
+	modelName := "gemini-pro-vision"
+
+	prompt := "What is in this video?"
+
```
Comment (eliben) [inline vertexai/multimodal-video/multimodal.go:36]: extra empty line?

**Comment by eliben** [comment]
> In this page we want to show a distinct snippet in the two section "Image" and "Video"
> 
> https://cloud.google.com/vertex-ai/docs/generative-ai/multimodal/send-multimodal-prompts#gemini-send-multimodal-samples-go

OK, fair enough, then please make it as identical as possible, except perhaps a comment and the link itself. I think there are some extra differences like whitespace right now

**Comment by Deleplace** [comment]
> OK, fair enough, then please make it as identical as possible, except perhaps a comment and the link itself. I think there are some extra differences like whitespace right now

Done, they're now line-for-line identical.  The difference is s/image/video/.

File: vertexai/multimodal-video/go.mod [hunk 2]
```diff
@@ -0,0 +1,34 @@
+module github.com/GoogleCloudPlatform/golang-samples/vertexai/multimodal-video
+
+go 1.21
+
+require cloud.google.com/go/vertexai v0.6.0
```
Comment (eliben) [inline vertexai/multimodal-video/go.mod:5]: Please update this to the latest, to align with https://github.com/GoogleCloudPlatform/golang-samples/pull/3776

Comment (Deleplace) [reply vertexai/multimodal-video/go.mod]: done

**Review by JulienBreux — APPROVED** [review]
LGTM

**Review by polong-lin — APPROVED** [review]

File: vertexai/multimodal/multimodal.go [hunk 3]
```diff
@@ -70,6 +71,10 @@ func generateMultimodalContent(w io.Writer, prompt, image, projectID, location,
 		return fmt.Errorf("unable to generate contents: %v", err)
 	}
 
+	if len(res.Candidates) == 0 || len(res.Candidates[0].Content.Parts) == 0 {
+		return fmt.Errorf("empty response from model")
```
Comment (teivah) [inline vertexai/multimodal/multimodal.go:75]:
```suggestion
		return errors.New("empty response from model")
```

File: vertexai/multimodal-video/multimodal.go [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+		FileURI:  video,
+	}
+
+	res, err := model.GenerateContent(ctx, img, genai.Text(prompt))
+	if err != nil {
+		return fmt.Errorf("unable to generate contents: %v", err)
+	}
+
+	if len(res.Candidates) == 0 || len(res.Candidates[0].Content.Parts) == 0 {
+		return fmt.Errorf("empty response from model")
```
Comment (teivah) [inline vertexai/multimodal-video/multimodal.go:75]:
```suggestion
		return errors.New("empty response from model")
```

**Review by teivah — APPROVED** [review]
LGTM

File: vertexai/multimodal-video/multimodalvideo.go [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+// generateMultimodalContent generates a response into w, based upon the prompt
+// and video provided.
+// video is a Google Cloud Storage path starting with "gs://"
+func generateMultimodalContent(w io.Writer, prompt, video, projectID, location, modelName string, temperature float32) error {
+	// prompt := "What is in this video?"
+	// video := "gs://cloud-samples-data/video/animals.mp4"
+	// location := "us-central1"
+	// modelName := "gemini-1.0-pro-vision"
+	// temperature := float32(0.4)
```
Comment (grayside) [inline vertexai/multimodal-video/multimodalvideo.go:38]: suggestion: The sample may benefit from hard-coding the temperature.

https://googlecloudplatform.github.io/samples-style-guide/#minimal-arguments

Comment (Deleplace) [reply vertexai/multimodal-video/multimodalvideo.go]: Thanks, done

**Pull request merged**